src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/GraphQLEndpointInitializerImpl.java

```diff
@@ -0,0 +1,281 @@
+        fieldPaths.add(typePath3 + ofTypePath + ofTypePath + iName);
+        fieldPaths.add(typePath3 + ofTypePath + ofTypePath + iKind);
+        fieldPaths.add(typePath3 + ofTypePath + ofTypePath + ofTypePath + iName);
+        fieldPaths.add(typePath3 + ofTypePath + ofTypePath + ofTypePath + iKind);
+
+        // ------------------------------------------------------
+        // Query type information for creating GraphQLEntrypoints
+        fieldPaths.add(queryTypePath1 + iName);
+        fieldPaths.add(queryTypePath2 + iName);
+        fieldPaths.add(queryTypePath3 + iName);
+        fieldPaths.add(queryTypePath3 + iKind);
+
+        // "Unwrap" query type
+        fieldPaths.add(queryTypePath3 + ofTypePath + iName);
+        fieldPaths.add(queryTypePath3 + ofTypePath + iKind);
+        fieldPaths.add(queryTypePath3 + ofTypePath + ofTypePath + iName);
+        fieldPaths.add(queryTypePath3 + ofTypePath + ofTypePath + iKind);
+        fieldPaths.add(queryTypePath3 + ofTypePath + ofTypePath + ofTypePath + iName);
+        fieldPaths.add(queryTypePath3 + ofTypePath + ofTypePath + ofTypePath + iKind);
+
+        // ------------------------------------------------------
+        // query type args
+        fieldPaths.add(queryTypeArgsPath1 + iName);
+        fieldPaths.add(queryTypeArgsPath2 + iName);
+        fieldPaths.add(queryTypeArgsPath2 + iKind);
+        fieldPaths.add(queryTypeArgsPath2 + ofTypePath + iName);
+        fieldPaths.add(queryTypeArgsPath2 + ofTypePath + iKind);
+
+        // ------------------------------------------------------
+        // Types to ignore when parsing and initializing types
+        fieldPaths.add(subscriptionPath + iName);
+        fieldPaths.add(mutationPath + iName);
+
+        return new GraphQLQueryImpl(fieldPaths, new HashSet<>());
+    }
+
+    /**
+     * Parses introspection data to determine information about a specific GraphQL field.
+     * @return a Pair consisting of the GraphQL valuetype (not including list and/or non-nullable identifiers) 
+     * and GraphQLFieldType respectively for the GraphQL type. @param field is a JsonObject and should 
+     * contain the keys: "name", "kind" and alternatively "ofType"
+     */
+    protected Pair<String,GraphQLFieldType> determineTypeInformation(final JsonObject field){
+
+        // ofTypePath segment without '/' at the end
+        final String ofTypeKey = ofTypePath.substring(0,ofTypePath.length() - 1);
+
+        // If current object is a wrapper type (i.e. LIST or NON-NULLABLE) use recursion
+        if(field.hasKey(ofTypeKey) && !field.get(ofTypeKey).isNull()){
+            return determineTypeInformation(field.getObj(ofTypeKey));
+        }
+
+        final String name = field.getString(iName);
+        final String kind = field.getString(iKind);
+        final GraphQLFieldType fieldType;
+        if(kind.equals("OBJECT")){
+            fieldType = GraphQLFieldType.OBJECT;
+        }
+        else{
```

Comment:
Oh, one small thing. Here, compare directly with "SCALAR" on an else if branch, and on the else branch throw an error.